src/Core/Math/Ellipsoid.js

```diff
@@ -1,182 +1,171 @@
- * @param {number} position.x
- * @param {number} position.y
- * @param {number} position.z
- * @param {Coordinate} [target] coordinate to copy result
- * @returns {Coordinate} an object describing the coordinates on the reference ellipsoid, angles are in degree
- */
-Ellipsoid.prototype.cartesianToCartographic = function cartesianToCartographic(position, target = new Coordinates('EPSG:4326', 0, 0, 0)) {
-    // for details, see for example http://www.linz.govt.nz/data/geodetic-system/coordinate-conversion/geodetic-datum-conversions/equations-used-datum
-    // TODO the following is only valable for oblate ellipsoid of revolution. do we want to support triaxial ellipsoid?
-    const R = Math.sqrt(position.x * position.x + position.y * position.y + position.z * position.z);
-    const a = this.rayon_1; // x
-    const b = this.rayon_3; // z
-    const e = Math.abs((a * a - b * b) / (a * a));
-    const f = 1 - Math.sqrt(1 - e);
-    const rsqXY = Math.sqrt(position.x * position.x + position.y * position.y);
-
-    const theta = Math.atan2(position.y, position.x);
-    const nu = Math.atan(position.z / rsqXY * ((1 - f) + e * a / R));
-
-    const sinu = Math.sin(nu);
-    const cosu = Math.cos(nu);
-
-    const phi = Math.atan((position.z * (1 - f) + e * a * sinu * sinu * sinu) / ((1 - f) * (rsqXY - e * a * cosu * cosu * cosu)));
-
-    const h = (rsqXY * Math.cos(phi)) + position.z * Math.sin(phi) - a * Math.sqrt(1 - e * Math.sin(phi) * Math.sin(phi));
-
-    return target.set('EPSG:4326',
-        THREE.Math.radToDeg(theta),
-        THREE.Math.radToDeg(phi),
-        h);
-};
+        normal.multiplyScalar(coordCarto.altitude());
 
-Ellipsoid.prototype.cartographicToCartesianArray = function cartographicToCartesianArray(coordCartoArray) {
-    var cartesianArray = [];
-    for (var i = 0; i < coordCartoArray.length; i++) {
-        cartesianArray.push(this.cartographicToCartesian(coordCartoArray[i]));
+        return target.add(normal);
     }
 
-    return cartesianArray;
-};
+    /**
+     * Convert cartesian coordinates to geographic according to the current ellipsoid of revolution.
+     * @param {Object} position - The coordinate to convert
+     * @param {number} position.x
+     * @param {number} position.y
+     * @param {number} position.z
+     * @param {Coordinate} [target] coordinate to copy result
+     * @returns {Coordinate} an object describing the coordinates on the reference ellipsoid, angles are in degree
+     */
+    cartesianToCartographic(position, target = new Coordinates('EPSG:4326', 0, 0, 0)) {
+        // for details, see for example http://www.linz.govt.nz/data/geodetic-system/coordinate-conversion/geodetic-datum-conversions/equations-used-datum
+        // TODO the following is only valable for oblate ellipsoid of revolution. do we want to support triaxial ellipsoid?
+        const R = Math.sqrt(position.x * position.x + position.y * position.y + position.z * position.z);
```

Comment:
```js
const R = position.length();
```

Reply:
It doesn't go through unit test, or I need to modify `Coordinates`; as I am currently rewriting this class in another PR, I will adapt this later, not here.